.github/workflows/ci.yml

```diff
@@ -1,35 +1,78 @@
-name: CI Build
+---
+name: Actions CI Pipeline
 on:
   pull_request:
-    branches: [main]
+    branches:
+      - main
   push:
-    branches: [main]
+    branches:
+      - main
 
 jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo ⤵️
+        uses: actions/checkout@v3
+
+      - name: Lint Dockerfile
+        uses: hadolint/hadolint-action@v2.0.0
+        with:
+          dockerfile: Dockerfile
+
+      - name: Lint kernel-crawler
+        uses: cclauss/GitHub-Action-for-pylint@0.7.0
+        with:
+          args: "pip install -r requirements.txt ; pylint kernel_crawler"
+        continue-on-error: true  # allow failure for now
+
+      - name: Lint YAML
+        uses: karancode/yamllint-github-action@v2.0.0
+        with:
+          yamllint_file_or_dir: '.github/workflows/ci.yml'
+          yamllint_strict: false
+          yamllint_comment: false
+
+  build:
+    name: Build
```

Comment:
It's ok but i'd use a path-filter to only build it if docker/ subfolder was changed. See https://github.com/falcosecurity/libs/blob/master/.github/workflows/ci.yml#L11 for an example :)

I'd also change the job name to "build Dockerfile"